code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -0,0 +1,131 @@
+/obj/machinery/smoke_machine
+	name = "Smoke Machine"
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "smoke0"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/smoke_machine
+	var/efficiency = 10
+	var/on = FALSE
+	var/cooldown = 0
+	var/screen = "home"
+	var/analyzeVars[0]
+	var/useramount = 30 // Last used amount
+	var/volume = 1000
+	var/setting = 3
+	var/list/possible_settings = list(3,6,9,12,15)
+
+/datum/effect_system/smoke_spread/chem/smoke_machine/set_up(datum/reagents/carry = null, setting = 3, efficiency = 10, loc)
+	amount = setting
+	carry.copy_to(chemholder, 20)
+	carry.remove_any(setting * 16 / efficiency)
+	location = loc
+
+/obj/machinery/smoke_machine/Initialize()
+	create_reagents(volume)
+	. = ..()
+/obj/machinery/smoke_machine/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/machinery/smoke_machine/RefreshParts()
+	efficiency = 6
+	for(var/obj/item/stock_parts/matter_bin/B in component_parts)
+		efficiency += B.rating
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
+		efficiency += C.rating
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		efficiency += M.rating
+
+/obj/machinery/smoke_machine/process()
+	..()
+	if(stat & NOPOWER)
+		icon_state = "smoke0"
+		update_icon()
```

Comment:
Put that icon_state setting in update_icon, why are you even calling it if you're doing it manually.